### DISCUSSION TITLE
Можно взять готовый конфиг из zapret от flowseal и добавить в этот?

### DISCUSSION THREAD

**Discussion by kilcha** [bol-van/zapret#787] [Q&A] [answered]
Возможно вопрос звучит крайне глупо, но если это можно сделать, то я бы хотел знать как. Параметры для tpws напоминают формат batника у flowseal. Всё это дело я проворачиваю на Linux с nftables.
Ещё последний вопрос. При установке задаются вопросы о включений tpws и nfqws, будет ли оно работать, если на оба вопроса ответить "Да"?

**Comment by bol-van** [comment, marked as answer]
> Существует 3 стандартных опции запуска, настраиваемых раздельно и независимо: `tpws-socks`, `tpws`, `nfqws`.
> Их можно использовать как по отдельности, так и вместе. Например, вам надо сделать комбинацию
> из методов, доступных только в `tpws` и только в `nfqws`. Их можно задействовать вместе.
> `tpws` будет прозрачно локализовывать трафик на системе и применять свое дурение, `nfqws` будет дурить трафик,
> исходящий с самой системы после обработки на `tpws`.
> А можно на эту же систему повесить без параметров socks proxy, чтобы получать доступ к обходу блокировок через прокси.
> Таким образом, все 3 режима вполне могут задействоваться вместе.
> Так же безусловно и независимо, в добавок к стандартным опциям, применяются все custom скрипты в `init.d/{sysv,openwrt,macos}/custom.d`.
> 
> Однако, при комбинировании tpws и nfqws не все так просто , как может показаться на первый взгляд.
> Первым всегда работает tpws, за ним - nfqws. На nfqws попадает уже "задуренный" трафик от tpws.
> Получается, что дурилка дурит дурилку, и дурилка не срабатывает, потому что ее задурили.
> Вот такой веселый момент. nfqws перестает распознавать протоколы и применять методы.
> Некоторые методы дурения от tpws nfqws в состоянии распознать и отработать корректно, но большинство - нет.
> Решение - использование `--dpi-desync-any-protocol` в nfqws и работа как с неизвестным протоколом.
> Комбинирование tpws и nfqws является продвинутым вариантом, требующим глубокого понимания происходящего.
> Очень желательно проанализировать действия nfqws по `--debug` логу. Все ли так, как вы задумали.
>

**Reply by bol-van** [reply]
Другой вариант - tpws на tcp, nfqws на udp. Или tpws на http, nfqws на https. Если разнести по протоколам и портам, вообще нет проблем

**Reply by kilcha** [reply]
Спасибо большое за ответ!
